AndroidSDKPush/src/main/java/com/leanplum/LeanplumNotificationHelper.java

```diff
@@ -195,6 +196,59 @@
     return builder;
   }
 
+  /**
+   * Gets NotificationCompat.Builder for provided parameters.
+   *
+   * @param context The application context.
+   * @param message Push notification Bundle.
+   * @param contentIntent PendingIntent.
+   * @param title String with title for push notification.
+   * @param messageText String with text for push notification.
+   * @param bigPicture Bitmap for BigPictureStyle notification.
+   * @param defaultNotificationIconResourceId int Resource id for default push notification icon.
+   * @return NotificationCompat.Builder or null.
+   */
+  static NotificationCompat.Builder getNotificationCompatBuilder(Context context, Bundle message,
+      PendingIntent contentIntent, String title, final String messageText, Bitmap bigPicture,
+      int defaultNotificationIconResourceId) {
+    NotificationCompat.Builder notificationCompatBuilder =
+        getNotificationCompatBuilder(context, message);
+    if (notificationCompatBuilder == null) {
+      return null;
+    }
+
+    if (defaultNotificationIconResourceId == 0) {
+      notificationCompatBuilder.setSmallIcon(context.getApplicationInfo().icon);
+    } else {
+      notificationCompatBuilder.setSmallIcon(defaultNotificationIconResourceId);
+    }
+
+    notificationCompatBuilder.setContentTitle(title)
+        .setStyle(new NotificationCompat.BigTextStyle()
+            .bigText(messageText))
+        .setContentText(messageText);
+
+    if (bigPicture != null) {
+      notificationCompatBuilder.setStyle(new NotificationCompat.BigPictureStyle()
+          .bigPicture(bigPicture)
+          .setBigContentTitle(title)
+          .setSummaryText(messageText));
+    }
+
+    // Try to put a notification on top of the notification area. This method was deprecated in API
+    // level 26. For API level 26 and above we must use setImportance(int) for each notification
+    // channel, not for each notification message.
+    if (Build.VERSION.SDK_INT >= 16 && !BuildUtil.isNotificationChannelSupported(context)) {
```

Comment:
16- minimum supported API level. Starting 26 it's deprecated, why we check for !BuildUtil.isNotificationChannelSupported(context) too. If it run on 26 with targetSDK 25 - it will works.

Reply:
got it.